resource.address/tcp/src/main/java/org/kaazing/gateway/resource/address/tcp/TcpResourceAddressFactorySpi.java

```diff
@@ -198,7 +210,8 @@ else if (bindAddress instanceof String) {
     @Override
     protected TcpResourceAddress newResourceAddress0(String original, String location) {
 
-        URI uriOriginal = URI.create(original);
+        // TODO: find appropriate implementation
+        URI uriOriginal = URI.create(location);
```

Comment:
As said earlier, we will use String for externalURI in ResourceAddress. So it will become
```
    public ResourceAddress(ResourceAddressFactorySpi factory, String externalURI, URI resourceURI) {
```

That way original can be passed as it is.